content/200-concepts/100-components/01-prisma-schema/09-index-configuration.mdx

```diff
@@ -129,17 +139,103 @@ model Post {
+
+## Full text indexes (MySQL and MongoDB)
+
+In version 3.6.0 and later, a new `@@fulltext` attribute is available as a preview feature. This provides support for introspection and migration of full text indexes in MySQL and MongoDB. Existing full text indexes in the database will be added to your Prisma schema after introspecting with `db push`, and new full text indexes added in the schema will be created in the database on migration. This also prevents validation errors in some database schemas that were not working before.
+
+For now we do not enable the full text search commands in the Prisma Client. They will be implemented separately, and the progress can be followed in the [MongoDB](https://github.com/prisma/prisma/issues/9413) and [MySQL](https://github.com/prisma/prisma/issues/10415) issues.
+
+### Enabling the preview feature
+
+To enable this preview feature, add it to the `generator` block of the `schema.prisma` file:
+
+```prisma file=schema.prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["fullTextIndex"]
+}
+```
+
+### Examples
+
+The following example demonstrates adding a `@@fulltext` index to the `title` and `content` fields of a `Post` model:
+
+```prisma file=schema.prisma
+model Post {
+  id      Int    @id
+  title   String @db.VarChar(255)
+  content String @db.Text
+
+  @@fulltext([title, content])
+}
+```
+
+On MongoDB, the `fullTextIndex` preview feature can be combined with the `extendedIndexes` preview feature to add ascending or descending columns to your full-text index:
+
+```prisma file=schema.prisma
+generator js {
+  provider        = "prisma-client-js"
+  previewFeatures = ["fullTextIndex", "extendedIndexes", "mongoDb"]
+}
+
+model Post {
+  id      String @id @map("_id") @db.ObjectId
+  title   String @db.VarChar(255)
+  content String @db.Text
+
+  @@fulltext([title(sort: Desc), content])
+}
+```
+
+### Upgrading from previous versions
+
+<Admonition type="warning">
+
+These are **breaking changes** when updating from certain existing databases. After updating, always run `prisma db pull` to introspect the existing database before migrating.
+
+</Admonition>
 
-In both of the cases above unwanted changes to your database can be prevented by properly specifying these properties in your data model where necessary. **The easiest way to do this is to use `prisma db pull` to retrieve the sort and length arguments where applicable.** Alternatively, you could also add these arguments manually. This should be done before using `prisma db push` or `prisma migrate dev`.
+Earlier versions of Prisma converted full text indexes using the `@@index` attribute rather than the `@@fulltext` attribute. After enabling the `fullTextIndex` preview feature, run `prisma db pull` to convert these indexes to `@@fulltext` before migrating.
```

Comment:
```suggestion
Earlier versions of Prisma converted full text indexes using the `@@index` attribute rather than the `@@fulltext` attribute. After enabling the `fullTextIndex` preview feature, run `prisma db pull` to convert these indexes to `@@fulltext` before migrating again with Prisma Migrate. If you do not do this, the existing indexes will be dropped instead and normal indexes will be created in their place.
```